plugins/modules/nmcli.py

```diff
@@ -2934,31 +2953,73 @@ def main():
 
         elif nmcli.state == "up":
             if nmcli.connection_exists():
-                if module.check_mode:
-                    module.exit_json(changed=True)
-                if nmcli.conn_reload:
-                    (rc, out, err) = nmcli.reload_connection()
-                (rc, out, err) = nmcli.up_connection()
-                if rc != 0:
-                    module.fail_json(name=f"Error bringing up connection named {nmcli.conn_name}", msg=err, rc=rc)
+                is_active = nmcli.is_connection_active()
+
+                if is_active and not nmcli.conn_reload:
+                    result["changed"] = False
+                    result["msg"] = f"Connection {nmcli.conn_name} is already active"
+                    if module.check_mode:
+                        module.exit_json(changed=False, **result)
+                else:
+                    if module.check_mode:
+                        module.exit_json(changed=True, **result)
+
+                    if nmcli.conn_reload:
+                        (rc, out, err) = nmcli.reload_connection()
+                        if rc != 0:
+                            module.fail_json(
+                                name=f"Error bringing up connection named {nmcli.conn_name}", msg=err, rc=rc
+                            )
+
+                    if not is_active or nmcli.conn_reload:
+                        (rc, out, err) = nmcli.up_connection()
+                        if rc != 0:
+                            module.fail_json(
+                                name=f"Error bringing up connection named {nmcli.conn_name}", msg=err, rc=rc
+                            )
+                        result["changed"] = True
+            else:
+                module.fail_json(
+                    name=nmcli.conn_name,
+                    msg="Connection does not exist",
+                )
 
         elif nmcli.state == "down":
             if nmcli.connection_exists():
-                if module.check_mode:
-                    module.exit_json(changed=True)
-                if nmcli.conn_reload:
-                    (rc, out, err) = nmcli.reload_connection()
-                (rc, out, err) = nmcli.down_connection()
-                if rc != 0:
-                    module.fail_json(name=f"Error bringing down connection named {nmcli.conn_name}", msg=err, rc=rc)
+                is_active = nmcli.is_connection_active()
+
+                if not is_active and not nmcli.conn_reload:
+                    result["changed"] = False
+                    result["msg"] = f"Connection {nmcli.conn_name} is already inactive"
+                    if module.check_mode:
+                        module.exit_json(changed=False, **result)
```

Comment:
same here - if there is nothing to be done, why return that only when check mode is on?